### ISSUE TITLE
Vnc Error - Problem Connecting

### ISSUE THREAD

**Issue by jsanchezzz** [neutrinolabs/xrdp#2086]
Hello, 

Im experiencing the following error with the tool xrdp in a suse 15 sp1 machine in azure, I just wanna know is someone has seen it before. 

Error: 

Please wait, we now perform access control...
Reply from access control: Sucess
connecting to sesman ip 127.0.0.01 port 3350
seman connect ok
sending login info to session manager, please wait....
login successful for display 201
VNC started connecting
waiting 2000 ms for VNC to start...
VNC connecting to 127.0.0.1 6101
VNC error - problem connecting
some problem

**Comment by matt335672** [comment]
@jsanchezzz - these two lines look odd:-
```
login successful for display 201
VNC connecting to 127.0.0.1 6101
```
Have you changed the `X11DisplayOffset=10` setting in sesman.ini? It looks like it's set to around 200, which probably won't work.

**Comment by matt335672** [comment]
@jsanchezzz - do you need any more help with this?

**Comment by jsanchezzz** [comment]
@matt335672 yes, I reinstalled the xrdp, but it showing the same error. Also I did not change the X11DisplayOffset=10, how can i see if the X11DisplayOffset has changed somehow?

**Comment by matt335672** [comment]
It's in `/etc/xrdp/sesman.ini`:-

```sh
$ sudo grep -i displayoffset /etc/xrdp/sesman.ini
;; X11DisplayOffset - x11 display number offset
X11DisplayOffset=10
```

**Comment by jsanchezzz** [comment]
Hello as you said the X11DisplayOffset is set to 200, and we dont know why? could you tell us how to change it to 10, please? see the image attached. 
![sesma 200](https://user-images.githubusercontent.com/96146459/153668747-636d42f5-6349-4368-b6cf-09df4ad04374.PNG)

**Comment by matt335672** [comment]
Well, you seem to have two problems here

The first problem is solved by editing `/etc/xrdp/sesman.ini`, correcting the value, and restarting the `xrdp-sesman` service.

The second and more serious problem is that you've got a file on your system which needs privilege to change and something unknown has changed it. That's not a good position to be in. Not only do we not know what else might have changed in the xrdp configuration, but there might be other files which have changed that could affect system operation. So I think it's worth trying to figure out how this file could have changed.

**Comment by jsanchezzz** [comment]
Hello @matt335672 

We changed the parameter but it did not work, We did a deeper investigation to know why the rdp is not working and we entered the logs, we saw something about a certificate, I would like to know if you have any knowledge, because the certificate is installed but the vm does not find it.

see the images attached and the notepad. 
[log JS.txt](https://github.com/neutrinolabs/xrdp/files/8062052/log.JS.txt)
![ujjyjuy](https://user-images.githubusercontent.com/96146459/153896772-2d91c9b7-895e-498c-a09a-d9e92eaa289b.PNG)
![jgjhj](https://user-images.githubusercontent.com/96146459/153896779-36b3c254-d69e-4d6b-b238-16e75ecd4867.PNG)

**Comment by matt335672** [comment]
If you don't have a certificate, xrdp will fall back on a less secure connection method, so I'd expect it still to work. I think you should leave that for now.

A couple of questions for you:-
- What version of xrdp is installed?
- What desktop have you installed on the machine?

**Comment by jsanchezzz** [comment]
Hello @matt335672 

What version of xrdp is installed?

Version 0.9.6

What desktop have you installed on the machine?

linux suse 15 sp1

**Comment by matt335672** [comment]
Thanks.

"Linux SuSE 15 sp 1" isn't what I'd call a desktop. I mean, are you running GNOME or KDE (or similar)? You'll need a desktop installed for xrdp to start. Do you have one installed?

As far as the certificate error goes

```
Cannot read certificate file /etc/xrdp/cert.pem: No such file or directory
```

I've checked the sources for 0.9.6 and it's telling me that either `/etc/xrdp` doesn't exist on that machine, or `/etc/xrdp/cert.pem` doesn't exist on that machine. What do you get on the machine for these two command when logged in as root:-

```
ls -l /etc/xrdp/cert.pem
openssl x509 -text -noout </etc/xrdp/cert.pem
```

**Comment by jsanchezzz** [comment]
Hello Matt, 

We solved the problem: 

1 - Performing a clean installation of patterns-gnome-gnome_basic
2 - Allowing remote sesion with remote manager
3 - Installing xrdp xorgxrdp
4 - enabling the vm firewall and open the port 3389 ( systemctl enable firewalld, systemctl startd firewalld firewall-cmd --permanent --zone=public --add-port=3389/tcp )

Thanks for you support, it  was rellay heplful to solve the issue. 

Regards.

**Comment by matt335672** [comment]
Thanks for letting us know.

Did you get your certificate problem sorted out too?

**Comment by frgomes** [comment]
Jumping into this conversation 3 years later.

I'm not asking a question here, but trying to add something to the quoted observation below:

> The second and more serious problem is that you've got a file on your system which needs privilege to change and something unknown has changed it. That's not a good position to be in. Not only do we not know what else might have changed in the xrdp configuration, but there might be other files which have changed that could affect system operation. So I think it's worth trying to figure out how this file could have changed.
--  matt335672

I've observed the same issue. The configuration at `/etc/xrdp/sesman.ini` says that the default is 10 but I see 200.
I'm not sure what changed this configuration.

However, I've observed something peculiar:

1. I'm running OpenSUSE Tumbleweed in a VM under Hyper-V under Windows11. Unfortunately I have to use Windows :-( 

2. I have GWSL, which is a Xserver with some tweaks intended to make life easier if you are using WSL. But I'm not using OpenSUSE under WSL; I'm using OpenSUSE under a VM, under Hyper-V.

3. Then I have two choices for opening a RDP session: (a) using GWSL or (b) using Remote Desktop Connection, which is native to Windows11.

4. I've discovered that GWSL employs/requires `DISPLAY=:10.0` whilst Windows Remote Desktop Connection employs/requires `DISPLAY=:200.0`

5. I can open KDE full screen using GWSL, but I have to open first a ssh session and then run `startplasma-x11` from it, which is inconvenient, but works.

6. However, I cannot open KDE full screen using Windows Remote Desktop Connection. But I've discovered that it starts to work fine if I first open a GWSL connection. In other words: if I open first a window with KDE in it via GWSL, after that, I can open another window running KDE in it via Windows Remote Desktop Connection. In this case, applications running in the first window employ `DISPLAY=:10.0` whilst application in the second window employ `DISPLAY=:200.0`.

Needless to say that I don't understand why it behaves this way.

Seems like the RDP server for display 10 works like a transport layer for the RDP server for display 200. The evidence is: if I close the RDP server for display 10 (i.e.: I close GWSL), the window containing applications associated to display 200 stop working after 10 seconds and the entire KDE session dies.

I would prefer to use WIndow Remote Desktop Server, since it performs better. But, as explained, it will require GWSL for some unknown reason... which forces me to use GWSL anyway.

Conclusion:

Trying to answer the question "who changed X11DisplayOffset=10 to X11DisplayOffset=200"... my guess is package `hyper-v`, installed automatically on OpenSUSE when the VM is first installed. I'm not sure though. Just a guess.
